tests/test_dynamodb2/test_dynamodb.py

```diff
@@ -2412,6 +2412,44 @@ def update(col, to, rv):
     with assert_raises(ClientError) as ex:
         r = update("col1", "val6", "WRONG")
 
+# https://github.com/spulec/moto/issues/3448
+@mock_dynamodb2
+def test_update_return_updated_new_attributes_when_same():
+    dynamo_client = boto3.resource("dynamodb", region_name="us-east-1")
+    dynamo_client.create_table(
+        TableName="moto-test",
+        KeySchema=[{'AttributeName': 'HashKey1', 'KeyType': 'HASH'}],
+        AttributeDefinitions=[{'AttributeName': 'HashKey1', 'AttributeType': 'S'}],
+        ProvisionedThroughput={'ReadCapacityUnits': 1, 'WriteCapacityUnits': 1},
+    )
+    
+
+    dynamodb_table = dynamo_client.Table("moto-test")
+    dynamodb_table.put_item(Item={
+        "HashKey1": "HashKeyValue1",
+        "listValuedAttribute1": ['a','b']})
+
+    def update(col, to, rv):
+        return dynamodb_table.update_item(
+            TableName="moto-test",
+            Key={"HashKey1": "HashKeyValue1"},
+            UpdateExpression=f"SET listValuedAttribute1=:{col}",
+            ExpressionAttributeValues={':'+col: to},
+            ReturnValues=rv,
+        )
+
+    r = update('a', ['a', 'c'], "UPDATED_NEW")
+    assert r["Attributes"] == {'listValuedAttribute1': ['a', 'c']}
+
+    r = update('a', {'a', 'c'}, "UPDATED_NEW")
+    assert r["Attributes"] == {'listValuedAttribute1': {'a', 'c'}}
+
+    r = update('a', {1,2}, "UPDATED_NEW")
+    assert r["Attributes"] == {'listValuedAttribute1': {1, 2}}
+
+    with assert_raises(ClientError) as ex:
```

Comment:
We've just changed to pytest instead of nose for our testing framework, so this will have to change to `with pytest.raises(..)`.
If you merge in the latest changes to master, you should the same change in the other tests